pallets/subspace/src/global.rs

```diff
@@ -99,6 +98,9 @@ impl<T: Config> Pallet<T> {
     }
 
     pub fn set_global_params(params: GlobalParams) {
+        // Check if the params are valid
+        Self::check_global_params(params.clone()).unwrap();
```

Comment:
I'm pretty sure `check_global_params` could receive `&GlobalParams` instead and avoid cloning it here.

Change it to an expect statement. Unwraps are usually unhelpful.
```suggestion
        Self::check_global_params(params.clone()).expect("global params are invalid");
```